### PR TITLE
[enhancement] add release workfolow for daily release and tag event.

### PR DESCRIPTION
**Pull request by rambohe-ch**
/kind documentation

#### What this PR does / why we need it:
Add a release workflow for updating images daily and updating images when a new tag is created.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

/assign @Congrool 

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      - name: Install Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+      - name: Release
+        run: make docker-push TARGET_PLATFORMS=linux/amd64,linux/arm64,linux/arm/v7 IMAGE_TAG=${{ steps.get_version.outputs.VERSION }}
```
Comment (Congrool) [inline .github/workflows/release.yaml:39]: It seems to should be `make docker-build` .

**Review by Congrool — COMMENTED** [review]
Seems some problems here:
If we are not at the tag, the image-tag should be `latest`, otherwise it should be `latest`. However, make rule will tag the image as `GIT_VERSION` which is based on the `IMAGE_TAG` and not be exposed to outside. Thus, what we set in `IMAGE_TAG` will not work.
Maybe we have to revise the makefile to make it work.

**Comment by rambohe-ch** [comment]
> Seems some problems here: If we are not at the tag, the image-tag should be the tag, otherwise it should be `latest`. However, make rule will tag the image as `GIT_VERSION` which is based on the `IMAGE_TAG` and not be exposed to outside. Thus, what we set in `IMAGE_TAG` will not work. Maybe we have to revise the makefile to make it work.

@rambohe-ch Thanks for your feedback. and I have exposed `IMAGE_TAG` as global env, so end user can set this env in the github action.

File: .github/workflows/release.yaml (hunk 1 above)
Comment (rambohe-ch) [reply .github/workflows/release.yaml:39]: @Congrool because i want to use `docker buildx build --push`, so `make docker-push` command has been added.

**Comment by Congrool** [comment]
>  Thanks for your feedback. and I have exposed IMAGE_TAG as global env, so end user can set this env in the github action.

Well, I mean `GIT_VERSION` is not exposed to the outside, it is the tag what images will actually use. You can find it here:
https://github.com/openyurtio/openyurt/blob/master/Makefile#L99

The arguments of `-t` option is `${IMAGE_REPO}/yurthub:${GIT_VERSION}`, not the ${IMAGE_TAG}. So, I think it may need to be revised.

**Comment by rambohe-ch** [comment]
> > Thanks for your feedback. and I have exposed IMAGE_TAG as global env, so end user can set this env in the github action.
> 
> Well, I mean `GIT_VERSION` is not exposed to the outside, it is the tag what images will actually use. You can find it here: https://github.com/openyurtio/openyurt/blob/master/Makefile#L99
> 
> The argument of `-t` option is `${IMAGE_REPO}/yurthub:${GIT_VERSION}`, not the ${IMAGE_TAG}. So, I think it may need to be revised.

@Congrool fixed it.

**Comment by Congrool** [comment]
/lgtm

It looks good to release image daily. But these logic in makefile will make it a little bit complicated and difficult to use. It's better the move them into a script. Maybe we can do it in further work.

**Pull request merged**